drivers/npmx/npmx_shell.c

```diff
@@ -688,1290 +696,1289 @@ static int cmd_adc_ntc_type_get(const struct shell *shell, size_t argc, char **a
+	if (!check_error_code(shell, err_code)) {
+		print_get_error(shell, "vout select");
 		return 0;
 	}
 
-	int err = 0;
-	uint8_t charger_set = CLAMP(shell_strtoul(argv[1], 0, &err), 0, UINT8_MAX);
+	print_value(shell, (int)vout_select, UNIT_TYPE_NONE);
+	return 0;
+}
 
-	if (err != 0) {
-		shell_error(shell, "Error: status has to be an integer.");
+static int cmd_charger_charging_current_set(const struct shell *shell, size_t argc, char **argv)
+{
+	args_info_t args_info = { .expected_args = 1,
+				  .arg = {
+					  [0] = { SHELL_ARG_TYPE_UINT32_VALUE, "charging current" },
+				  } };
+	if (!arguments_check(shell, argc, argv, &args_info)) {
 		return 0;
 	}
 
-	if (charger_set > 1) {
-		shell_error(shell, "Error: wrong new status value.");
+	npmx_charger_t *charger_instance = charger_instance_get(shell, "charging current", true);
+	if (charger_instance == NULL) {
 		return 0;
 	}
 
-	npmx_charger_t *charger_instance = npmx_charger_get(npmx_instance, 0);
-	npmx_error_t (*change_status_function)(const npmx_charger_t *p_instance,
-					       uint32_t module_mask) =
-		charger_set == 1 ? npmx_charger_module_enable_set : npmx_charger_module_disable_set;
-
-	npmx_error_t err_code = change_status_function(charger_instance, mask);
-
-	if (check_error_code(shell, err_code)) {
-		shell_print(shell, "Success: %d.", charger_set);
-	} else {
-		shell_error(shell, "Error: unable to set charging module status.");
+	uint16_t charging_current_ma = CLAMP(args_info.arg[0].result.uvalue, 0, UINT16_MAX);
+	npmx_error_t err_code =
+		npmx_charger_charging_current_set(charger_instance, charging_current_ma);
+	if (!check_error_code(shell, err_code)) {
+		print_set_error(shell, "charging current");
+		return 0;
 	}
 
+	print_success(shell, charging_current_ma, UNIT_TYPE_MILLIAMPERE);
 	return 0;
 }
 
-static int cmd_charger_module_get(const struct shell *shell, size_t argc, char **argv,
-				  npmx_charger_module_mask_t mask)
+static int cmd_charger_charging_current_get(const struct shell *shell, size_t argc, char **argv)
 {
 	ARG_UNUSED(argc);
 	ARG_UNUSED(argv);
 
-	npmx_instance_t *npmx_instance = npmx_driver_instance_get(pmic_dev);
+	npmx_charger_t *charger_instance = charger_instance_get(shell, NULL, false);
+	if (charger_instance == NULL) {
+		return 0;
+	}
 
-	if (npmx_instance == NULL) {
-		shell_error(shell, "Error: shell is not initialized.");
+	uint16_t charging_current_ma;
+	npmx_error_t err_code =
+		npmx_charger_charging_current_get(charger_instance, &charging_current_ma);
+	if (!check_error_code(shell, err_code)) {
+		print_get_error(shell, "charging current");
 		return 0;
 	}
 
-	npmx_charger_t *charger_instance = npmx_charger_get(npmx_instance, 0);
-	uint32_t module_mask;
+	print_value(shell, charging_current_ma, UNIT_TYPE_MILLIAMPERE);
+	return 0;
+}
 
-	npmx_error_t err_code = npmx_charger_module_get(charger_instance, &module_mask);
+static int charger_die_temp_set(const struct shell *shell, size_t argc, char **argv,
```

Comment:
Charger changes should be in a separate commit

Reply:
Fixed.